txinput.go

```diff
@@ -2,14 +2,31 @@ package bt
 
 import (
 	"bytes"
+	"context"
 	"encoding/binary"
+	"errors"
 	"fmt"
 
 	"github.com/libsv/go-bk/crypto"
 
 	"github.com/libsv/go-bt/v2/bscript"
 )
 
+// FundIteratorFunc is used for AutoFund. It expect bt.FundIteration to be returned
+// containing relevant input information, and a bool informing if the retrieval was successful.
+//
+// It is expected that the boolean return value should return false once funds are depleted.
+type FundIteratorFunc func() (*FundIteration, bool)
+
+// FundIteration contains information relating to the current fund. Its fields are intended
+// for use with tx.From(...).
+type FundIteration struct {
```

Comment:
Rename this to just Fund I think - does that clash with anything else?

Reply:
Nope, but I'm concious of `bt.Fund` being confused with `tx.Input`. What do you reckon?

Reply:
Maybe it should just return a bt.Input, any reason it wouldn't?

Reply:
Not really, it just means the the implementor has to build the Input themselves, being, parse the lockingscript into a *bscript.Script, and manually add the input tx id, instead of just being able to call the `From` helper function.

I'll change it over here now.